### PR TITLE
Document password hash additions to documentation.

### PR DESCRIPTION
**Pull request by SandPod**
Should not be merged until we release version 1.2.6.

Documents the peppering and secure random configurations.

### CODE REVIEW

File: docs/05-concepts/10-authentication/04-providers/01-email.md [hunk 1]
```diff
@@ -121,3 +121,39 @@ await authController.resetPassword(email, verificationCode, password);
...
+To configure a pepper, set the `emailPasswordPepper` property in the `config/passwords.yaml` file.
+
+```yaml
+development:
+  emailPasswordPepper: 'your-pepper'
+```
+
+ It is essential to keep the pepper secret and never expose it to the client.
+
+ If the pepper is changed, all passwords in the database will need to be re-hashed with the new pepper.
```
Comment (Isakdl) [inline docs/05-concepts/10-authentication/04-providers/01-email.md:145]: Suggestion Make this into an info box or maybe even a warning box?

```md
:::info
If the pepper is changed, all passwords in the database will need to be re-hashed with the new pepper.
:::

```

File: docs/05-concepts/10-authentication/04-providers/01-email.md [hunk 2]
```diff
@@ -121,3 +121,39 @@ await authController.resetPassword(email, verificationCode, password);
...
+
+It is possible to prevent this fallback by setting the `allowUnsecureRandom` property in the `AuthConfig` to `false`. If the `allowUnsecureRandom` property is false, the server will throw an exception if a secure random number generator is unavailable.
+
+```dart
+auth.AuthConfig.set(auth.AuthConfig(
+  allowUnsecureRandom: false,
+));
+```
+
+The default value will change to `false` in Serverpod 2.0.
```
Comment (Isakdl) [inline docs/05-concepts/10-authentication/04-providers/01-email.md:159]: Suggestion: This should already be false on main (if not we should change it). Meaning this sentence can already be removed from the `next` version.

**Review by Isakdl — CHANGES_REQUESTED** [review]
Looking good, but I have two suggested improvements!

File: docs/05-concepts/10-authentication/04-providers/01-email.md (hunk 2 above)
Comment (SandPod) [reply docs/05-concepts/10-authentication/04-providers/01-email.md:159]: Good call!

PR to change the default here: https://github.com/serverpod/serverpod/pull/2022

**Review by Isakdl — APPROVED** [review]
LGTM

**Pull request merged**